src/engine/naive_engine.cc

```diff
@@ -159,7 +160,11 @@ class NaiveEngine final : public Engine {
     this->req_completed_ = false;
     profiler::Profiler *profiler = profiler::Profiler::Get();
     NaiveOpr *opr = nullptr;
+    // Try to get a new name. Only not null for sub-operators of a custom operator
     const bool profiling = opr_name && profiler->IsProfiling(profiler::Profiler::kImperative);
+    const char* d_name = profiling && opr_name ?
+                         profiler::CustomOpProfiler::Get()->GenerateDisplayName(opr_name) : nullptr;
+    const char* display_name = (d_name ? d_name : opr_name);
     if (profiling) {
       opr = NewOperator(exec_fun, const_vars, mutable_vars,
```

Comment:
Who frees this ptr?

Reply:
This printer is retuned by std::string.c_str(). The memory that it points to is managed by the original string and will be deallocated as it goes out of scope

Reply:
string's are stored in the singleton CustomOpProfiler object

Reply:
I meant opr

Reply:
That's a great question. I tried to look it up and I also could not find where the deletion took place

Reply:
Is the d_name here really necessary? Can we simplify the logic here to just the display_name?

Reply:
its done by DeleteOperator

Reply:
I think we need to keep it. Because we need to check if the returned char* is nullptr or not

Reply:
How about this?
```suggestion
const char* display_name = profiling ? profiler::CustomOpProfiler::Get()->GenerateDisplayName(opr_name) : opr_name
```

Then you move the handling of opr_name == nullptr to inside GenerateDisplayName(opr_name). i.e., when opr_name is nullptr, this function also returns a nullptr.